### PR TITLE
Feature/sync greeting and ready

### PR DESCRIPTION
**Pull request by voldemortman**

### CODE REVIEW

File: src/createStreamingManager.ts [hunk 1]
```diff
@@ -76,7 +85,11 @@ function pollStats(peerConnection: RTCPeerConnection, onVideoStateChange) {
 
             if (!isStreaming) {
                 onVideoStateChange?.(StreamingState.Start);
-
+                if (streamsCount >= streamsBeforeReady && !getIsConnected()) {
+                    console.log('connecting from video stream')
+                    connect();
```
Comment (benyael15) [inline src/createStreamingManager.ts:90]: why connect?

**Review by benyael15 — COMMENTED** [review]
review with @HaMMeRSI

File: src/createStreamingManager.ts (hunk 1 above)
Comment (benyael15) [reply src/createStreamingManager.ts:89]: call is connectionCallback
its more clear when connect that at first i tought is the actul connetion api - but its not
it is just a callback and flag

File: src/createAgentManager.ts [hunk 2]
```diff
@@ -118,6 +118,7 @@ function initializeStreamAndChat(
                 callbacks: {
                     ...options.callbacks,
                     onConnectionStateChange: async state => {
+                        options.callbacks.onConnectionStateChange?.(state);
```
Comment (HaMMeRSI) [inline src/createAgentManager.ts:121]: whyd you moved it here?

connected is only happens after new chat started

File: src/createAgentManager.ts [hunk 3]
```diff
@@ -326,7 +324,7 @@ export async function createAgentManager(agent: string, options: AgentManagerOpt
         async function connectWithRetry() {
             for (let attempt = 1; attempt <= maxRetries; attempt++) {
                 try {
-                    return await initializeStreamAndChat(agentInstance, options, agentsApi, analytics, items.chat, greeting);
+                    return await initializeStreamAndChat(agentInstance, options, agentsApi, analytics, items.chat, ...(newChat ? [greeting] : []));
```
Comment (HaMMeRSI) [inline src/createAgentManager.ts:327]: aint there a prettier way of doing this?
move this logic to getGreeting()?

File: src/createStreamingManager.ts [hunk 4]
```diff
@@ -58,13 +59,21 @@ function createVideoStatsAnalyzer() {
     };
 }
 
-function pollStats(peerConnection: RTCPeerConnection, onVideoStateChange) {
+function pollStats(
+    peerConnection: RTCPeerConnection,
+    onVideoStateChange,
+    warmup: boolean = false,
+    getIsConnected: () => boolean,
+    connect: () => void) {
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:67]: onConnect

**Review by HaMMeRSI — CHANGES_REQUESTED** [review]

**Pull request merged**